src/data_write.py

```diff
@@ -8,175 +8,213 @@
 
     :copyright: 2017 SuperDARN Canada
 """
-
-import sys
-import os
-import datetime
-import json
+# built-in
+import argparse as ap
 import collections
-import warnings
-import time
-import threading
+import copy
+import datetime
+from dataclasses import dataclass, field, fields
 import errno
+import faulthandler
+import json
 from multiprocessing import shared_memory
+import os
+import pickle
 import subprocess as sp
-import argparse as ap
+import sys
+import threading
+import time
+from typing import List, Dict, Set
+import warnings
+
+# third-party
 import numpy as np
 import deepdish as dd
 import tables
 import zmq
-import faulthandler
 from scipy.constants import speed_of_light
-import copy
-import pickle
-import utils.options.data_write_options as dwo
+
+# local
 from utils import socket_operations as so
+from utils.message_formats import ProcessedSequenceMessage
+import utils.options.data_write_options as dwo
 
 
-DATA_TEMPLATE = {
-    "borealis_git_hash"         : None, # Identifies the version of Borealis that made this data.
-    "experiment_id"             : None, # Number used to identify experiment.
-    "experiment_name"           : None, # Name of the experiment file.
```

Comment:
Should we keep these comments? Or are these available elsewhere in our documentation?

Reply:
The description comments should be added to the metadata dict under “description” or something. This can then be parsed to build the docs. In this way, we only have one place where changes need to be made.

Reply:
All of these fields are documented in the respective data files that they belong in, so it doesn't seem necessary to me to repeat that here. We don't use the fields, we're just moving them around.

Reply:
> The description comments should be added to the metadata dict under “description” or something. 

I don't know how easy it will be to pull them into the docs from this dataclass, let alone format them nicely, but I like the idea of putting these comments into the metadata field.